### PR TITLE
Removing '%h' and '%b' from 'dbfilter' to avoid the wrong detection database (issue #58).

### PR DESCRIPTION
**Pull request by xt0ph**

### CODE REVIEW

File: connector/queue/worker.py [hunk 1]
```diff
@@ -263,6 +263,8 @@ def available_db_names():
             db_names = db.exp_list(True)
         dbfilter = config['dbfilter']
         if dbfilter and db_names:
+            # Remove %d and %h from dbfilter (issue #58)
+            dbfilter = re.sub('(%d|%h).*$', '', dbfilter)
             db_names = [d for d in db_names if re.match(dbfilter, d)]
```
Comment (sbidoul) [inline connector/queue/worker.py:268]: Does it make sense to match on dbfilter after removing %d and %h?

As @guewen noted in #58 it probably does not make sence to use dbfilter at all in the job runner.

So we could completely remove this filtering, or to be less brutal towards user who would depend on this, skip the filtering if dbfilter contains %d or %h.

**Comment by xt0ph** [comment]
Except if the database name contains a fixed part referencing the instance. For example dbfilter contains the 'o80_%d', so I need the connector reference the database beginning with 'o80_'  but not all databases on the server.

**Comment by sbidoul** [comment]
But if dbfilter is 'odoo_%d_80', then removing the %d makes no sense. So your situation works a bit by chance, no?

**Comment by xt0ph** [comment]
Not by chance, but at the origin of the '%d' is part of the research, it will not run. We have a prefix on our bases so it works. This is not an ideal solution but it works much better than the original. If someone has a better idea for the filter works with the '%d' (in all cases) I'm interested.

**Comment by sbidoul** [comment]
I now think the connector should completely ignore dbfilter. The very fact that a client hostname can be part of the filter proves it's not a concept to be used by the connecter which is completely independent from clients.

So as a first step I would write 

``` python
if dbfilter and '%d' not in dbfilter and '%h' not in dbfilter:
    db_names = [d for d in db_names if re.match(dbfilter, d)]
```

So it does not break with dbfilters containing %d or %h and it continues to work with other dbfilters until we deprecate it's use completely in the connector.

**Comment by guewen** [comment]
Thanks @2cadz for the proposal.
I agree with @sbidoul. Please see #165 for the proposed correction path.

**Pull request closed without merging**